torch/_dynamo/variables/tensor.py

```diff
@@ -256,13 +260,31 @@ def unpack_var_sequence(self, tx, idxes=None):
             idxes = range(length)
         return [wrap_fx_proxy(tx, self.as_proxy()[i], **options) for i in idxes]
 
+    def _strict_mode_banned_ops(self):
+        return [
+            "stride",
+            "requires_grad",
+            "storage_offset",
+            "device",
+            "layout",
+            "is_cuda",
+            "is_quantized",
+            "is_meta",
+            "data",
+            "is_sparse",
+            "ndim",
```

Comment:
ndim is fine, torch.compile specializes on the number of dimensions and we do know ahead of time how many dimensions the grads have.

If we want to make this more robust we could blanket ban any method that starts with `is_`, because if you're querying this you are probably about to switch on it.

Reply:
ack